### PR TITLE
Model rework part 3 electric boogalee

### PR DESCRIPTION
**Pull request by EGKangaroo**

### CODE REVIEW

**Review by tsan21 — APPROVED** [review]

File: src/Controllers/SolutionsController.cs [hunk 1]
```diff
@@ -76,23 +78,57 @@ public async Task<IActionResult> PutSolution(Guid id, Solution solution)
...
+            foreach (var item in dataModel.SDGs)
+            {
+                if (!SDGExists(item.SDGId))
+                {
+                    return NotFound();
+                }
+            }
+
+            //ensure many to many relationships are properly defined
+            await PopulateManyToManySDGs(dataModel);
```
Comment (LarsVDriel) [inline src/Controllers/SolutionsController.cs:124]: Use _context.AttachRange(howTo.SDGs) to do this

File: src/Converters/ArticleRequestModelConverter.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                UploadDate = DateTime.Now,
+                LastUpdatedTime = DateTime.Now,
+                ViewCount = 0,
+                Content = requestModel.Content
+            };
+
+            //Note: the linked many-to-many data still needs to be processed at a later stage!
+            foreach(var sdg in requestModel.SDGIds)
+            {
+                article.SDGs.Add(new SDGSolution() { SDGId = sdg });
```
Comment (LarsVDriel) [inline src/Converters/ArticleRequestModelConverter.cs:30]: Use _context.AttachRange(article.SDGs) to do this

File: src/Converters/HowToRequestModelConverter.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                AuthorId = requestModel.AuthorId,
+                UploadDate = DateTime.Now,
+                LastUpdatedTime = DateTime.Now,
+                ViewCount = 0
+            };
+
+            //Note: the linked many-to-many data still needs to be processed at a later stage!
+            foreach (var sdg in requestModel.SDGIds)
+            {
+                howTo.SDGs.Add(new SDGSolution() { SDGId = sdg });
```
Comment (LarsVDriel) [inline src/Converters/HowToRequestModelConverter.cs:31]: Use _context.AttachRange(howTo.SDGs) to do this

File: src/Converters/HowToRequestModelConverter.cs [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+            }
+
+            foreach(var tool in requestModel.Tools)
+            {
+                howTo.Tools.Add(new Tool() { Name = tool });
```
Comment (LarsVDriel) [inline src/Converters/HowToRequestModelConverter.cs:36]: Use _context.AttachRange(howTo.Tools) to do this

File: src/Converters/HowToRequestModelConverter.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+            }
+
+            foreach(var material in requestModel.Materials)
+            {
+                howTo.Materials.Add(new Material() { Name = material });
```
Comment (LarsVDriel) [inline src/Converters/HowToRequestModelConverter.cs:41]: Use _context.AttachRange(howTo.Materials) to do this

File: src/Converters/HowToRequestModelConverter.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            }
+
+            foreach(var step in requestModel.Steps)
+            {
+                howTo.Steps.Add(new Step() { CoverImage = step.CoverImage, Description = step.Description });
```
Comment (LarsVDriel) [inline src/Converters/HowToRequestModelConverter.cs:46]: Use _context.AttachRange(howTo.Steps) to do this

**Review by LarsVDriel — CHANGES_REQUESTED** [review]

File: src/Controllers/SolutionsController.cs (hunk 1 above)
Comment (EGKangaroo) [reply src/Controllers/SolutionsController.cs:124]: https://docs.microsoft.com/en-us/ef/core/saving/related-data

Dat hoeft niet

File: src/Controllers/SolutionsController.cs [hunk 7]
```diff
@@ -76,23 +78,57 @@ public async Task<IActionResult> PutSolution(Guid id, Solution solution)
...
+            foreach(var item in dataModel.SDGs)
+            {
+                if (!SDGExists(item.SDGId))
+                {
+                    return NotFound();
+                }
+            }
+
+            //ensure many to many relationship is properly defined
+            await PopulateManyToManySDGs(dataModel);
```
[the remark this answers is not in the document]
Comment (EGKangaroo) [reply src/Controllers/SolutionsController.cs:96]: https://docs.microsoft.com/en-us/ef/core/saving/related-data

dat hoeft niet

**Review by LarsVDriel — APPROVED** [review]

**Pull request merged**